src/lib/network_pool/transaction_pool.ml

```diff
@@ -2677,4 +2677,32 @@ let%test_module _ =
           [%test_eq: pool_apply] (accepted_commands apply_res) (Ok all_cmds) ;
           assert_pool_txs all_cmds ;
           Deferred.unit)
+
+    (* TODO: Where should these tests go that need block_on_async_exn? *)
```

Comment:
@mrmr1993 any idea on where this test should go? mina_base doesn't have access to `Thread_safe` afaik

Reply:
@bkase The test can stay in parties.ml if we use `Run_in_thread.block_on_async_exn`, because downstream libraries can decide if they import `run_in_thread.native` or `run_in_thread.fake`. Will update this PR one final time with this change